File: src/types/supplier-debt.ts
export interface SupplierDebt {
  id: number;
  supplier_id: number;
  supplier_name: string;
  purchase_id: number | null;
  total_amount: number;
  paid_amount: number;
  remaining_amount: number;
  due_date: string | null;
  status: string;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface SupplierPayment {
  id: number;
  debt_id: number;
  amount: number;
  payment_date: string;
  notes: string | null;
  recorded_by: number;
  created_at: string;
}

export interface SupplierDebtDetail {
  debt: SupplierDebt;
  payments: SupplierPayment[];
}

export interface SupplierDebtListItem {
  id: number;
  supplier_id: number;
  supplier_name: string;
  purchase_id: number | null;
  total_amount: number;
  paid_amount: number;
  remaining_amount: number;
  due_date: string | null;
  status: string;
  notes: string | null;
  created_at: string;
}

export interface CreateSupplierDebtRequest {
  supplier_id: number;
  purchase_id?: number;
  total_amount: number;
  paid_amount?: number;
  due_date?: string;
  notes?: string;
}

export interface RecordSupplierPaymentRequest {
  debt_id: number;
  amount: number;
  payment_date?: string;
  notes?: string;
  recorded_by: number;
}

File: src-tauri/src/models/supplier_debt.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SupplierDebt {
    pub id: i64,
    pub supplier_id: i64,
    pub supplier_name: String,
    pub purchase_id: Option<i64>,
    pub total_amount: f64,
    pub paid_amount: f64,
    pub remaining_amount: f64,
    pub due_date: Option<String>,
    pub status: String,
    pub notes: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SupplierPayment {
    pub id: i64,
    pub debt_id: i64,
    pub amount: f64,
    pub payment_date: String,
    pub notes: Option<String>,
    pub recorded_by: i64,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateSupplierDebtRequest {
    pub supplier_id: i64,
    pub purchase_id: Option<i64>,
    pub total_amount: f64,
    pub paid_amount: Option<f64>,
    pub due_date: Option<String>,
    pub notes: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RecordSupplierPaymentRequest {
    pub debt_id: i64,
    pub amount: f64,
    pub payment_date: Option<String>,
    pub notes: Option<String>,
    pub recorded_by: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SupplierDebtListItem {
    pub id: i64,
    pub supplier_id: i64,
    pub supplier_name: String,
    pub purchase_id: Option<i64>,
    pub total_amount: f64,
    pub paid_amount: f64,
    pub remaining_amount: f64,
    pub due_date: Option<String>,
    pub status: String,
    pub notes: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SupplierDebtDetail {
    pub debt: SupplierDebt,
    pub payments: Vec<SupplierPayment>,
}
